cfn-resources/federated-database-instance/mongodb-atlas-federateddatabaseinstance.json

```diff
@@ -0,0 +1,355 @@
+        },
+        "delete": {
+            "permissions": ["secretsmanager:GetSecretValue"]
+        },
+        "list": {
+            "permissions": ["secretsmanager:GetSecretValue"]
+        }
+    },
+    "properties": {
+        "CloudProviderConfig": {
+            "type": "object",
+            "description": "Cloud provider linked to this data lake.",
+            "$ref": "#/definitions/CloudProviderConfig"
+        },
+        "DataProcessRegion": {
+            "type": "object",
+            "description": "Information about the cloud provider region to which the data lake routes client connections. MongoDB Cloud supports AWS only.",
+            "$ref": "#/definitions/DataProcessRegion"
+        },
+        "ProjectId": {
+            "type": "string",
+            "description": "Unique 24-hexadecimal digit string that identifies your project.",
+            "maxLength": 24,
+            "minLength": 24,
+            "pattern": "^([a-f0-9]{24})$"
+        },
+        "TenantName": {
+            "type": "string",
+            "description": "Human-readable label that identifies the data federation."
+        },
+        "SkipRoleValidation": {
+            "type": "boolean",
+            "description": "Flag that indicates whether this request should check if the requesting IAM role can read from the S3 bucket. AWS checks if the role can list the objects in the bucket before writing to it. Some IAM roles only need write permissions. This flag allows you to skip that check."
+        },
+        "Storage": {
+            "type": "object",
+            "description": "Configuration information for each data store and its mapping to MongoDB Cloud databases.",
+            "$ref": "#/definitions/Storage"
+        },
+        "State": {
+            "type": "string",
+            "description": "Type of Federated Database Instances to return."
+        },
+        "HostNames": {
+            "type": "array",
+            "insertionOrder": false,
+            "description": "Type of Federated Database Instances to return.",
+            "items": {
+                "type": "string"
+            }
+        },
+        "Profile": {
+            "type": "string",
+            "description": "The profile is defined in AWS Secret manager. See [Secret Manager Profile setup](../../../examples/profile-secret.yaml).",
+            "default": "default"
+        }
+    },
+    "readOnlyProperties": [
+        "/properties/CloudProviderConfig/ExternalId",
+        "/properties/CloudProviderConfig/IamAssumedRoleARN",
+        "/properties/CloudProviderConfig/IamUserARN",
+        "/properties/HostNames",
+        "/properties/State",
+        "/properties/TenantName"
+    ],
+    "writeOnlyProperties": [
+        "/properties/CloudProviderConfig/TestS3Bucket"
+    ],
+    "createOnlyProperties": [
+        "/properties/ProjectId",
+        "/properties/Profile"
+    ],
+    "required": [
+        "ProjectId"
```

Comment:
remember to add all the required properties, on the CreateRequiredFields i see more, we need to add them here, because that affects the CDK

Reply:
Sure